### PR TITLE
set gaussian and blur by cfg

### PR DESCRIPTION
**Pull request by liujiaxing7**

### CODE REVIEW

File: src/utils.c [hunk 1]
```diff
@@ -798,6 +798,23 @@ int rand_int(int min, int max)
     int r = (random_gen()%(max - min + 1)) + min;
     return r;
 }
+//return a random odd number in an interval
+int rand_int_odd(int min, int max)
+{
+    if (max < min){
```
Comment (smartadpole) [inline src/utils.c]: 不要复制粘贴

File: src/utils.c [hunk 2]
```diff
@@ -798,6 +798,23 @@ int rand_int(int min, int max)
...
+        int s = min;
+        min = max;
+        max = s;
+    }
+    int r = (random_gen()%(max - min + 1)) + min;
+    if(r % 2){
```
Comment (smartadpole) [inline src/utils.c]: 用位移实现取奇数

File: src/parser.c [hunk 3]
```diff
@@ -1260,6 +1260,23 @@ void parse_net_options(list *options, network *net)
     }
     else fprintf(stderr, " GPU isn't used \n");
 #endif// GPU
+
+    char *g = option_find(options, "gaussian_noise_boundary");
```
Comment (smartadpole) [inline src/parser.c:1264]: 命名知名达意

File: src/option_list.c [hunk 4]
```diff
@@ -121,6 +121,26 @@ char *option_find_str_quiet(list *l, char *key, char *def)
     return def;
 }
 
+
+int *option_get_boundary(char *g)
```
Comment (smartadpole) [inline src/option_list.c]: 不要自己创造新代码，仿照原有数据的读取方式实现；比如学习率变化轮数是怎么读取进来的

File: src/utils.c (hunk 2 above)
Comment (liujiaxing7) [reply src/utils.c]: ok

File: src/option_list.c (hunk 4 above)
Comment (liujiaxing7) [reply src/option_list.c]: 这个就是仿照学习率变化轮数得来的
       char *l = option_find(options, "steps");
        char *p = option_find(options, "scales");
        char *s = option_find(options, "seq_scales");
        if(net->policy == STEPS && (!l || !p)) error("STEPS policy must have steps and scales in cfg file", DARKNET_LOC);

        if (l) {
            int len = strlen(l);
            int n = 1;
            int i;
            for (i = 0; i < len; ++i) {
                if (l[i] == '#') break;
                if (l[i] == ',') ++n;
            }
            int* steps = (int*)xcalloc(n, sizeof(int));
            float* scales = (float*)xcalloc(n, sizeof(float));
            float* seq_scales = (float*)xcalloc(n, sizeof(float));
            for (i = 0; i < n; ++i) {
                float scale = 1.0;
                if (p) {
                    scale = atof(p);
                    p = strchr(p, ',') + 1;
                }
                float sequence_scale = 1.0;
                if (s) {
                    sequence_scale = atof(s);
                    s = strchr(s, ',') + 1;
                }
                int step = atoi(l);
                l = strchr(l, ',') + 1;
                steps[i] = step;
                scales[i] = scale;
                seq_scales[i] = sequence_scale;
            }

File: src/utils.c (hunk 1 above)
Comment (liujiaxing7) [reply src/utils.c]: 这里也需要增加一个这样的判断

File: src/parser.c (hunk 3 above)
Comment (liujiaxing7) [reply src/parser.c:1264]: 仿照steps哪里的命名,以参数首个小字母命名

**Pull request merged**